modules/service/src/main/scala/lucuma/odb/graphql/mapping/LookupFrom.scala

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.odb.graphql
+package mapping
+
+import grackle.Path
+import grackle.Type
+
+trait LookupFrom[F[_]] extends BaseMapping[F] {
+
+  private lazy val StepRecordTypes: List[Type] =
+    List(
+      StepRecordType,
+      GmosNorthStepRecordType,
+      GmosSouthStepRecordType
+    )
+
+  private lazy val VisitTypes: List[Type] =
+    List(
+      VisitType,
+      GmosNorthVisitType,
+      GmosSouthVisitType
+    )
+
+  private def defineLookup(tpes: List[Type], path: Seq[String], mapping: ObjectMapping): List[(Path, ObjectMapping)] =
+    tpes.toList.map { tpe =>
+      (path.foldLeft(Path.from(tpe)) { case (t, p) => t / p }, mapping)
+    }
+
+  def lookupFromStepRecord(mapping: ObjectMapping, path: String*): List[(Path, ObjectMapping)] =
+    defineLookup(StepRecordTypes, path, mapping)
+
+  def lookupFromVisit(mapping: ObjectMapping, path: String*): List[(Path, ObjectMapping)] =
```

Comment:
Not particularly pleased by this, but wherever there is a `SwitchMapping` from `Visit` (in this case), we have to define the same mapping from each of the interface type `Visit` and the two concrete types `GmosNorthVisitType` and `GmosSouthVisitType`.  Since this is needed multiple times it seemed important to avoid repeating the type list.